server/src/main/java/org/apache/druid/server/router/ManualTieredBrokerSelectorStrategy.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.router;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Optional;
+import org.apache.commons.lang.StringUtils;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.query.Query;
+import org.apache.druid.query.QueryContexts;
+
+import javax.annotation.Nullable;
+
+/**
+ * Implementation of {@link TieredBrokerSelectorStrategy} which uses the parameter
+ * {@link QueryContexts#BROKER_SERVICE_NAME} in the Query context to select the
+ * Broker Service.
+ * <p>
+ * If the {@link #defaultManualBrokerService} is set to a valid Broker Service Name,
+ * then all queries that do not specify a valid value for
+ * {@link QueryContexts#BROKER_SERVICE_NAME} would be directed to the
+ * {@code #fallbackBrokerService}. Note that the {@code fallbackBrokerService}
```

Comment:
What is `fallbackBrokerService`? Is it the same as `defaultManualBrokerService`?

Reply:
Yes, it is the same. I have fixed the name.